### PR TITLE
Add button to trace for 30 seconds

### PR DESCRIPTION
**Pull request by akalin-keybase**
The idea is that users can click this button and the resulting trace
file will be included in a log send (in a future PR), or if testing,
a developer can pull it from the phone's filesystem.

### CODE REVIEW

File: shared/settings/advanced/index.js [hunk 1]
```diff
@@ -33,11 +34,47 @@ const Advanced = (props: Props) => (
...
+class TraceButton extends React.Component<TraceButtonProps, TraceButtonState> {
+  state = {
+    enabled: true,
+  }
+
+  _onClick = () => {
+    this.setState({enabled: false})
+    this.props.onTrace(this.props.durationSeconds)
+    setTimeout(() => {
+      this.setState({enabled: true})
```
Comment (chrisnojima) [inline shared/settings/advanced/index.js]: this is not safe. use the timerhoc to wrap this component

**Comment by chrisnojima** [comment]
can i see an example of this output in a logsend or just a file

File: shared/settings/advanced/index.js (hunk 1 above)
Comment (akalin-keybase) [reply shared/settings/advanced/index.js]: Changed to do the delay in the saga and to store the state in state.waiting, so that it survives tab changes

**Comment by akalin-keybase** [comment]
This file isn't included in log sends yet, so here's an example file:

[Keybase.trace.out.zip](https://github.com/keybase/client/files/1709318/Keybase.trace.out.zip)

Unzip it and view it using `go tool trace /path/to/Keybase.trace.out`.

This one has a lot of events because I blew away my local DB just to see what the activity would look like.

It's a binary file, but looking at the docs [here](https://docs.google.com/document/d/1FP5apqzBgr7ahCCgFO-yoVhk4YZrNIDNf9RybngBc14/edit) I don't see any sensitive info. All the stack traces are arrays of program counters, so that's just the functions that are called.

PTAL!

**Pull request merged**